### ISSUE TITLE
`datetime.datetime` and `datetime.date` should be incompatible, but aren't

### ISSUE THREAD

**Issue by tomasaschan** [microsoft/pyright#1240]
**Describe the bug**
`datetime.datetime` and `datetime.date` cannot be used interchangeably, but Pyright only reports issues in one direction; `datetime.datetime` is OK where a `datetime.date` is required.

**To Reproduce**
```python
import datetime

def foo(d: datetime.date):
    bar(d)  # correctly reports type error

def bar(dt: datetime.datetime):
    foo(dt)  # expected error, but no issue reported
```

**Expected behavior**
I expect type errors to be reported both for `foo(dt)` and `bar(d)`, but no issue is reported for `foo(dt)`.

**VS Code extension or command-line**
Problem shows up in both Pyright console tool version 1.1.84, Pylance extension version v2020.12.0.

**Comment by erictraut** [comment]
Pyright has no intrinsic knowledge of types like `date` or `datetime`. It bases its analysis on the type information provided by type stubs. For stdlib types like `date` and `datetime`, these type stubs come from the [typeshed repo](https://github.com/python/typeshed). The `datetime.pyi` typeshed stub indicates that [`datetime` is a subclass of `date`](https://github.com/python/typeshed/blob/3d8a29705e939fc626c09381d36fdc380e041577/stdlib/2and3/datetime.pyi#L222), which means that an object of type `datetime` is also a valid `date`. If you think this is incorrect, please file a bug or a PR in the typeshed repo. Based on the information provided to pyright, it's doing the right thing here.
